azure/services/privatedns/privatedns.go

```diff
@@ -59,7 +61,15 @@ func (s *Service) Reconcile(ctx context.Context) error {
 	if zoneSpec != nil {
 		// Create the private DNS zone.
 		s.Scope.V(2).Info("creating private DNS zone", "private dns zone", zoneSpec.ZoneName)
-		err := s.client.CreateOrUpdateZone(ctx, s.Scope.ResourceGroup(), zoneSpec.ZoneName, privatedns.PrivateZone{Location: to.StringPtr(azure.Global)})
+		pDNS := privatedns.PrivateZone{
+			Location: to.StringPtr(azure.Global),
+			Tags: converters.TagsToMap(infrav1.Build(infrav1.BuildParams{
+				ClusterName: s.Scope.ClusterName(),
+				Lifecycle:   infrav1.ResourceLifecycleOwned,
+				Additional:  s.Scope.AdditionalTags(),
+			})),
+		}
+		err := s.client.CreateOrUpdateZone(ctx, s.Scope.ResourceGroup(), zoneSpec.ZoneName, pDNS)
```

Comment:
Wouldn't this cause us to add the tags to any pre-existing DNS zone, including a user's BYO zone?

Reply:
This is good point. I was thinking on the similar lines. But when I looked at piublicip reconcile there was no such handling in the Reconcile for managed vs unmanaged. ( Not sure if it applies to that or not ) 
Ref: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/azure/services/publicips/publicips.go#L83
But I will make the changes here so that BYO DNS does not get tagged ( and hence reconciled) by CAPZ

Reply:
Interesting, we should verify but I think that might be a bug. Also coincidentally would be fixed by https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1745/files#diff-101a2cb7fa509c5208b45ff64bc92af67c470f1a3de730a5a05f60b95b7b1681R58.

Reply:
BTW the only issue with not tagging existing DNS zones is that we won't tag the ones created by earlier versions of CAPZ and thus it would cause the old DNS zones to be treated as BYO and not be deleted with this new code.

Reply:
added a comment in https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1692

Reply:
>BTW the only issue with not tagging existing DNS zones is that we won't tag the ones created by earlier versions of CAPZ and thus it would cause the old DNS zones to be treated as BYO and not be deleted with this new code.

Shall we cover this as part of upgrade? One of the things that I can think top off my head is, before upgrading CAPZ users can annotate DNS to adopt it for reconcile. May be we can automate some of this process but there has to be an user intervention unless the DNS that is NOT a BYO has some footprint that differentiates to that of DNS that is a BYO.

Reply:
Infact, IMO we can make a generic annotation key to solve similar issue for other resources or in general if we at some point of time want to adopt resources that are not reconciled or BYO 

What do you think?

Reply:
>BTW the only issue with not tagging existing DNS zones is that we won't tag the ones created by earlier versions of CAPZ and thus it would cause the old DNS zones to be treated as BYO and not be deleted with this new code.

This would only affect a subset of clusters where resource group is also not managed. Perhaps, we can add some logging for those scenarios to nudge the user in the right direction?

Reply:
Sure. I am already logging for not managed DNS.

Reply:
let's change that log to v(1) so it's printed even if the user doesn't change verbosity and add a little blurb about adding the tag to nudge the user to add the tag if they didn't create the zone themselves? Maybe we can even point them to a link in docs if we document this feature